### PR TITLE
Performance bugs

### PR DESCRIPTION
**Pull request by benghaziboy**
@MichaelButkovic @mparent61

### CODE REVIEW

File: tests/contributions/test_contributions_utils.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+
+class MergeRoleTestCase(BaseIndexableTestCase):
+
+    def setUp(self):
+        super(MergeRoleTestCase, self).setUp()
+        user_cls = get_user_model()
+        self.dominant = ContributorRole.objects.create(name="Dominator")
+        self.deprecated = ContributorRole.objects.create(name="Deprecated")
+        self.fella = user_cls.objects.create(first_name="fella", last_name="guy")
```
Comment (MichaelButkovic) [inline tests/contributions/test_contributions_utils.py]: fella

**Comment by MichaelButkovic** [comment]
LGTM

File: tests/contributions/test_contributions_utils.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+        self.feature_types = [
+            FeatureType.objects.create(name="surf"),
+            FeatureType.objects.create(name="turf"),
+            FeatureType.objects.create(name="smurf"),
+            FeatureType.objects.create(name="burf")
```
Comment (mparent61) [inline tests/contributions/test_contributions_utils.py]: burf???

Comment (benghaziboy) [inline tests/contributions/test_contributions_utils.py]: my mother's maiden name

File: bulbs/contributions/utils.py [hunk 3]
```diff
@@ -4,6 +4,64 @@
...
+            for flat_override in override.override_flatrate.all():
+                flat_override.profile = dom_override
+                flat_override.save()
+            for hourly_override in override.override_hourly.all():
+                hourly_override.profile = dom_override
+                hourly_override.save()
+            for feature_type_override in override.override_feature_type.all():
+                feature_type_override.profile = dom_override
+                feature_type_override.save()
+
```
Comment (mparent61) [inline bulbs/contributions/utils.py]: When does the deprecated role get deleted in normal workflow?

File: tests/contributions/test_contributions_utils.py (hunk 2 above)
Comment (mparent61) [inline tests/contributions/test_contributions_utils.py]: Lowe-Burf sorta sounds like "someone of low birth"

**Comment by mparent61** [comment]
:+1:

File: tests/contributions/test_contributions_utils.py (hunk 2 above)
Comment (benghaziboy) [inline tests/contributions/test_contributions_utils.py]: my caste

File: bulbs/contributions/utils.py (hunk 3 above)
Comment (benghaziboy) [inline bulbs/contributions/utils.py]: I don't want it deleted as part of this migration in the event of failure

**Pull request merged**